graylog2-server/src/main/java/org/graylog/plugins/views/search/views/ViewResolver.java

```diff
@@ -44,4 +46,22 @@
      * {@link org.graylog.plugins.views.search.db.SearchesCleanUpJob}.
      */
     Set<String> getSearchIds();
+
+    /**
+     * A method to return whether the current user is authorized to read the current view and its related objects
+     * (such as the backing {@link org.graylog.plugins.views.search.Search} record).
+     *
+     * This method accepts two permissions tester predicates, to allow each implementation to perform any combination
+     * of permission checks needed.
+     *
+     * @param viewId                  The id of the view.
+     * @param permissionTester        A predicate, which tests a single string permission parameter.
+     *                                For example, is the user authorized to read a view based on having of a
+     *                                particular permission?
+     * @param entityPermissionsTester A bi-predicate which tests two parameters (the permission to test and the id of
+     *                                the entity). For example, is the user authorized to read a view based on having
+     *                                a particular permission and/or having access to a particular entity?
+     */
+    boolean canReadView(String viewId, Predicate<String> permissionTester,
+                        BiPredicate<String, String> entityPermissionsTester);
```

Comment:
The addition of this method, allows view resolvers to provide their own specific permissions check as appropriate for the view view resolver. Both the single permission and permission/id predicates are passed as arguments, to allow the view resolver to decide which specific permissions/combination to check.

Passing the permission check predicates in also eliminates the need to inject the security principal into the view resolver implementation to check the permission.